MyAddin/Form1.cs

```diff
@@ -0,0 +1,317 @@
+        {
+            try
+            {
+                if (treeView1.SelectedNode == null)
+                {
+                    MessageBox.Show("Please Select a Node to add new node.", "Error", MessageBoxButtons.OK);
+                }
+
+                else if (name1 == "")
+                {
+                    MessageBox.Show("Please Enter name of Node to add new node.", "Error", MessageBoxButtons.OK);
+                }
+                else
+                {
+                    TreeNode node = treeView1.SelectedNode;
+                    node.Nodes.Add(name1);
+                }
+            }
+            catch (Exception)
+            {
+                MessageBox.Show("Please Select a Node to add new node.", "Error", MessageBoxButtons.OK);
+            }
+            treeView1.SelectedNode = null;
+            textBox1.Clear();
+            return;
+        }
+
+        //here user give name for new node
+        private void textBox1_TextChanged(object sender, EventArgs e)
+        {
+            try
+            {
+                name1 = textBox1.Text;  //get name from user for new node
+            }
+            catch (Exception)
+            {
+                MessageBox.Show("Not a Valid Name.", "Error", MessageBoxButtons.OK);
+            }
+        }
+
+        //this button contain functionality to remove node
+        private void button2_Click(object sender, EventArgs e)
+        {
+            try
+            {
+                treeView1.SelectedNode.Remove();
+            }
+            catch (Exception)
+            {
+                MessageBox.Show("Please Select a Node to remove.", "Error", MessageBoxButtons.OK);
+            }
+            treeView1.SelectedNode = null;
+            return;
+        }
+
+        //this button contain functionality to reset treeview
+        private void button3_Click(object sender, EventArgs e)
+        {
+            try
+            {
+                treeView1.Nodes.Clear();
+                textBox1.Clear();
+                GenericFTS(); //else generic FTS loaded
+                treeView1.ExpandAll();
+            }
+            catch (Exception)
+            {
+                MessageBox.Show("Unable to reset.", "Error", MessageBoxButtons.OK);
+            }
+        }
+
+        ////this button contain functionality to close addin
+        private void button4_Click(object sender, EventArgs e)
+        {
+            try
+            {
+                if (MessageBox.Show("Confirm?", "Close Application", MessageBoxButtons.YesNo) == DialogResult.Yes)
+                {
+                    this.Close();
+                }
+                else
+                {
+                    this.Activate();
+                }
+            }
+            catch (Exception)
+            {
+                MessageBox.Show("Problem Closing the Application.", "Error", MessageBoxButtons.OK);
+                treeView1.Nodes.Clear();
```

Comment:
If you have to write the same block  of code 2 or more times, move them to a new function.